sympy/stats/symbolic_probability.py

```diff
@@ -132,6 +150,47 @@ def doit(self, **hints):
 
         return self
 
+    def doit(self, **hints):
+        deep = hints.get('deep', True)
+        expr = self.args[0]
+        condition = self._condition
+        evaluate = self._evaluate
+        numsamples = hints.get('numsamples', False)
+        for_integral = hints.get('for_integral', False)
+        if not (evaluate or deep or for_integral):
+            return self
+        if deep:
+            expr = expr.doit(**hints)
+        if not random_symbols(expr) or isinstance(expr, Expectation):  # expr isn't random?
+            return expr
+        if numsamples:  # Computing by monte carlo sampling?
+            evalf = hints.get('evalf', True)
+            return sampling_E(expr, condition, numsamples=numsamples, evalf=evalf)
+
+        if expr.has(RandomIndexedSymbol):
+            return pspace(expr).compute_expectation(expr, condition, evaluate)
+
+        # Create new expr and recompute E
+        if condition is not None:  # If there is a condition
+            return self.func(given(expr, condition), evaluate=evaluate).doit(**hints)
+
+        # A few known statements for efficiency
+
+        if expr.is_Add:  # We know that E is Linear
+            return Add(*[self.func(arg, evaluate=evaluate).doit(**hints)
```

Comment:
Code partly repeating some logic from `.expand()`. Can we find a way to reuse the code? I.e., suppose the expansion will become more complicated in the future, it would be nice for `.doit()` to share the logic with `.expand()`. But it's not important, this way is also good.

Reply:
There is a slight difference while handling nested expectations. Th